src/main/java/com/github/javachat/doublerange/DoubleRange.java

```diff
@@ -0,0 +1,240 @@
+     * <p>
+     * It follows the semantics laid out in it's
+     * <a href="http://google.github.io/guava/releases/19.0/api/docs/com/google/common/collect/Range.html#encloses(com.google.common.collect.Range)">
+     * guava equivalent</a>.
+     */
+    public boolean encloses(DoubleRange other) {
+        return this.equals(other) // Enclosure is reflexive.
+                || (lowerCheck.test(other.lowerBound)
+                && lowerCheck.test(other.upperBound)
+                && upperCheck.test(other.lowerBound)
+                && upperCheck.test(other.upperBound));
+    }
+
+    /**
+     * Returns true if there exists a (possibly empty) range which is enclosed by both this range and other.
+     */
+    public boolean isConnected(DoubleRange other) {
+        return (this.lowerCheck.test(other.upperBound) && this.upperCheck.test(other.upperBound))
+                || (this.upperCheck.test(other.lowerBound) && this.lowerCheck.test(other.lowerBound));
+    }
+
+    public DoubleRange intersection(DoubleRange other) {
+        if (!this.isConnected(other)) {
+            throw new IllegalArgumentException(String.format(NO_CONNECTION, this, other));
+        }
+
+        return new DoubleRange(
+                DoubleCut.max(this.toLowerCut(), other.toLowerCut()),
+                DoubleCut.min(this.toUpperCut(), other.toUpperCut())
+        );
+    }
+
+    public DoubleRange span(DoubleRange other) {
+        return new DoubleRange(
+                DoubleCut.min(this.toLowerCut(), other.toLowerCut()),
+                DoubleCut.max(this.toUpperCut(), other.toUpperCut())
+        );
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(lowerBound, lowerBoundType, upperBound, upperBoundType);
+    }
+
+    @SuppressWarnings("OverlyComplexBooleanExpression")
+    @Override
+    public boolean equals(final Object obj) {
+        if (this == obj)
+            return true;
+        if (!(obj instanceof DoubleRange)) // also takes care of obj == null
+            return false;
+        final DoubleRange other = (DoubleRange) obj;
+        return lowerBound == other.lowerBound
+                && upperBound == other.upperBound
+                && lowerBoundType == other.lowerBoundType
+                && upperBoundType == other.upperBoundType;
+    }
+
+    @Override
+    public String toString() {
+        @SuppressWarnings("StringBufferReplaceableByString")
+        final StringBuilder sb = new StringBuilder();
+
+        sb.append(lowerBoundType == BoundType.CLOSED ? "[" : "(");
+        sb.append(TO_STRING_FORMAT.format(lowerBound)).append("..").append(TO_STRING_FORMAT.format(upperBound));
+        sb.append(upperBoundType == BoundType.CLOSED ? "]" : ")");
+
+        return sb.toString();
+    }
+
+    public static DoubleRange open(double lowerBound, double upperBound) {
+        return new DoubleRange(lowerBound, BoundType.OPEN, upperBound, BoundType.OPEN);
```

Comment:
Maybe some static factory methods .open() and .closed() could be added to DoubleCut?